### DISCUSSION TITLE
How to serialize data to xml using element namespaces?

### DISCUSSION THREAD

**Discussion by stefknecht** [FirelyTeam/firely-net-sdk#1985] [Q&A] [answered]
I didn't find a way how to serialize data to xml using element namespaces.

Is there any way to do this?

Example:

My used code:
//info: xml = xml-string of "original-xml-data"
var parsedData = parser.Parse<Bundle>(xml);

FhirXmlSerializer serializer = new FhirXmlSerializer(new SerializerSettings { Pretty = true });

string xmlText = serializer.SerializeToString(parsedData);

In xmlText the elements are only shown without the namespace ("xmlns"), except of the root element ("Bundle")

Example of original-xml-data:
`<Bundle xmlns="http://hl7.org/fhir">
      <Composition xmlns="http://hl7.org/fhir"></Composition >
      <Patient xmlns="http://hl7.org/fhir"></Patient>
</Bundle>`

Data written by "serializer.SerializeToString":
`<Bundle xmlns="http://hl7.org/fhir">
      <Composition></Composition >
      <Patient></Patient>
</Bundle>`

**Comment by mbaltus** [comment, marked as answer]
The FHIR specification mentions this on the [XML format page](http://hl7.org/fhir/xml.html):
"FHIR elements are always in the namespace http://hl7.org/fhir. This is usually specified as the default namespace on the root element."
The SDK is implemented according to that, so it just puts the namespace on the top level element when serializing the data.
